scripts/validations.py

```diff
@@ -0,0 +1,125 @@
+import itertools
+import json
+import sys
+
+ACTION_CONTEXT = 'context'
+ACTION_COUNT = 'count'
+ACTIONS = (ACTION_CONTEXT, ACTION_COUNT)
+
+UNLISTED_ADDONS = set()
+
+
+def parse_validations(results):
+    return (json.loads(result) for result in results)
+
+
+def unlisted_validations(results, unlisted_addons=None):
+    if unlisted_addons is None:
+        unlisted_addons = UNLISTED_ADDONS
+    return (result
+            for result in results
+            if (not result['metadata'].get('listed', True)
+                or result['metadata'].get('id') in unlisted_addons))
```

Comment:
Looks like there is overlap. Ideally we'd only get `(True, False, False)` and `(False, False, True)` in the results. We don't see any occurrences of `(True, False, False)` and we see 15% of validations in the `(False, False, True)` case.

* 40% of validations that have listed=False in metadata __do not__ have a matching unlisted addon.
* 100% of validations with listed=True or listed=None __do__ have a matching unlisted addon.

```
((listed, id is None, id in UNLISTED_ADDONS), count)
((None, False, True), 1243)
((False, False, False), 151)
((False, False, True), 254)
((False, True, False), 15)
((True, False, True), 22)
```

I'm quite surprised by these results. Test code: https://gist.githubusercontent.com/mstriemer/7ff3617b95866fb0ba84/raw/92d76c222c99c3bb27247fe5b0c380935c9a461b/-